src/SFML.Audio/SoundBuffer.cs

```diff
@@ -181,13 +180,17 @@ public Time Duration
         /// (sf::Int16).
         /// </summary>
         ////////////////////////////////////////////////////////////
-        public short[] Samples
+        public Span<short> Samples
         {
             get
             {
-                short[] SamplesArray = new short[sfSoundBuffer_getSampleCount(CPointer)];
-                Marshal.Copy(sfSoundBuffer_getSamples(CPointer), SamplesArray, 0, SamplesArray.Length);
-                return SamplesArray;
+                ulong count = sfSoundBuffer_getSampleCount(CPointer);
+                IntPtr ptr = sfSoundBuffer_getSamples(CPointer);
+
+                unsafe
+                {
+                    return new Span<short>((void*)ptr, (int)count);
```

Comment:
Since we .NET doesn't own the memory for samples, should we be handling this this way?  Our `Span` could be invalidated if CSFML moves things IIRC

Reply:
This will be safe for the lifetime of the SoundBuffer object instance for as long as all public members on SoundBuffer class don't produce iterator invalidation on the underlying `std::vector` containing the samples on `sf::SoundBuffer` (m_samples).